crates/ruff_python_semantic/src/node.rs

```diff
@@ -1,57 +1,65 @@
 use std::ops::{Index, IndexMut};
 
 use ruff_index::{newtype_index, IndexVec};
-use ruff_python_ast::Stmt;
 use rustc_hash::FxHashMap;
 
 use ruff_python_ast::types::RefEquality;
 
-/// Id uniquely identifying a statement in a program.
+/// Id uniquely identifying an AST node in a program.
 ///
 /// Using a `u32` is sufficient because Ruff only supports parsing documents with a size of max `u32::max`
-/// and it is impossible to have more statements than characters in the file. We use a `NonZeroU32` to
+/// and it is impossible to have more nodes than characters in the file. We use a `NonZeroU32` to
 /// take advantage of memory layout optimizations.
 #[newtype_index]
 #[derive(Ord, PartialOrd)]
 pub struct NodeId;
 
-/// A [`Node`] represents a statement in a program, along with a pointer to its parent (if any).
+/// A [`Node`] represents an AST node in a program, along with a pointer to its parent (if any).
 #[derive(Debug)]
-struct Node<'a> {
-    /// The statement this node represents.
-    stmt: &'a Stmt,
+struct Node<'a, T> {
+    /// A pointer to the AST node.
+    node: &'a T,
```

Comment:
Should we use `AnyNodeRef` instead?

Reply:
Hm, but I want the struct to be generic on nodes of a specific type (e.g., an `IndexVec` for statements, and one for expressions, rather than an `IndexVec` that can hold a mixture of nodes). Is that achievable with `AnyNodeRef`?